containers/payments/subscription/BlackFridayModal.js

```diff
@@ -0,0 +1,190 @@
+import PropTypes from 'prop-types';
+import {
+    classnames,
+    FormModal,
+    Loader,
+    Button,
+    Alert,
+    CurrencySelector,
+    useLoading,
+    useApi,
+    Price
+} from 'react-components';
+import { checkSubscription } from 'proton-shared/lib/api/payments';
+import { CYCLE, DEFAULT_CURRENCY, DEFAULT_CYCLE } from 'proton-shared/lib/constants';
+import { c } from 'ttag';
+
+const { MONTHLY, YEARLY, TWO_YEARS } = CYCLE;
+
+import './BlackFridayModal.scss';
+
+const BlackFridayModal = ({ bundles = [], onSelect, ...rest }) => {
+    const api = useApi();
+    const [loading, withLoading] = useLoading();
+    const [currency, updateCurrency] = useState(DEFAULT_CURRENCY);
+    const [pricing, updatePricing] = useState({});
+
+    const DEAL_TITLE = {
+        [MONTHLY]: c('Title').t`1 month deal`,
+        [YEARLY]: c('Title').t`1 year deal`,
+        [TWO_YEARS]: c('Title').t`2 year deal`
+    };
+
+    const BILLED_DESCRIPTION = ({ cycle, amount, notice }) =>
+        ({
+            [MONTHLY]: c('Title').jt`Billed as ${amount} (${notice})`,
+            [YEARLY]: c('Title').jt`Billed as ${amount} (${notice})`,
+            [TWO_YEARS]: c('Title').jt`Billed as ${amount} (${notice})`
+        }[cycle]);
+
+    const AFTER_INFO = ({ cycle, amount, notice }) =>
+        ({
+            [MONTHLY]: c('Title')
+                .jt`(${notice}) After one month, your subscription will automatically renew at the regular price of ${amount} every month.`,
+            [YEARLY]: c('Title')
+                .jt`(${notice}) After one year, your subscription will automatically renew at the regular price of ${amount} every year.`,
+            [TWO_YEARS]: c('Title')
+                .jt`(${notice}) After two years, your subscription will automatically renew at the regular price of ${amount} every two years.`
+        }[cycle]);
+
+    const getBundlePrices = async () => {
+        const result = await Promise.all(
+            bundles.map(({ planIDs = [], cycle = DEFAULT_CYCLE, couponCode }) => {
+                return Promise.all([
+                    api(
+                        checkSubscription({
+                            PlanIDs: planIDs,
+                            CouponCode: couponCode,
+                            Currency: currency,
+                            Cycle: cycle
+                        })
+                    ),
+                    api(
+                        checkSubscription({
+                            PlanIDs: planIDs,
+                            Currency: currency,
+                            Cycle: MONTHLY
+                        })
+                    )
+                ]);
+            })
+        );
+
+        updatePricing(
+            result.reduce((acc, [withCoupon, withoutCouponMonthly], index) => {
+                acc[index] = {
+                    withCoupon: withCoupon.Amount + withCoupon.CouponDiscount,
+                    withoutCoupon: withCoupon.Amount,
```

Comment:
`withoutCoupon: withCoupon.Amount` . Just a bit counter-intuitive, but I guess it's ok